02-express-tutorial/app.js

```diff
@@ -1 +1,41 @@
-console.log('Express Tutorial')
+const express = require('express')
+const { products } = require("./data");
+
+const app = express();
+
+const port = 3000;
+
+app.use(express.static("./public"))
+
+app.get('/api/v1/test', (req, res) => {
+    res.json({ message: 'It worked!' })
+})
+
+app.get('/api/v1/products', (req, res) => {
+    res.json(products)
+})
+
+app.get('/api/v1/products/:productID', (req, res) => {
+    const id = parseInt(req.params.productID)
+    const findId = products.find((item) => item.id === id)
+    if (!findId) {
+        res.status(404).send('Product not found')
+    } else {
+        res.json(findId)
+    }
+})
+
+app.get('/api/v1/query', (req, res) => {
+    const search = req.query.search
+    const cost = req.query.cost
+    const filteredProducts = products.filter((items) => items.search === search && items.cost === limit)
```

Comment:
You have to fix this logic here because it isn't working.
The parameter search you need to use to filter the products that name starts with search parameter value received.
The logic could be like this: 
`const filteredProducts = products.filter((item) => item.name.startsWith(search))`
You also need to check if the parameter search is defined. You can use an IF for doing that.
The limit parameter you need to use to filter the number of records that you need to return.
The logic could be like this:
`filteredProducts = filteredProducts.slice(0, Number(limit))`
Again you need to check if the limit parameter is defined because it could not be sent.